File: 백준/Bronze/2884. 알람 시계/알람 시계.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        String x = scan.nextLine();
        String[] num = x.split(" ");
        int hour = Integer.parseInt(num[0]);
        int minute = Integer.parseInt(num[1]);
        
        minute -= 45;
        if(minute < 0){
            minute = 60 + minute;
            hour -= 1;
            if(hour == -1){
                hour = 23;
            }
        } 
        System.out.printf("%d %d", hour, minute);
        scan.close();
    }
}

File: 백준/Bronze/2884. 알람 시계/알람 시계.py
H, M = map(int, input().split(' '))

if M - 45 >= 0:
    M -= 45
elif H == 0:
    H = 23
    M = 15 + M
else:
    H -= 1
    M = 15 + M
print(f"{H} {M}")